listenbrainz_spark/tests/test_utils.py

```diff
@@ -1,14 +1,96 @@
+import os
 import unittest
+from datetime import datetime
 
-from listenbrainz_spark import utils
+import listenbrainz_spark
+from listenbrainz_spark import utils, hdfs_connection
 from listenbrainz_spark.tests import SparkTestCase
 
+from pyspark.sql import Row
+
+HDFS_CLUSTER_URI = 'hdfs://hadoop-master:9000'
+
 class UtilsTestCase(SparkTestCase):
 
+    def test_append_dataframe(self):
+        path_ = 'test_df.parquet'
+        hdfs_path = os.path.join(HDFS_CLUSTER_URI, path_)
+
+        df = utils.create_dataframe(Row(column1=1, column2=2), schema=None)
+        utils.append(df, hdfs_path)
+        new_df = utils.read_files_from_HDFS(hdfs_path)
+        self.assertEqual(new_df.count(), 1)
+
+        df = utils.create_dataframe(Row(column1=3, column2=4), schema=None)
+        utils.append(df, hdfs_path)
+        appended_df = utils.read_files_from_HDFS(hdfs_path)
+        self.assertEqual(appended_df.count(), 2)
+        # delete_dir dir in every test func so that using identical dirs by chance will not throw error.
```

Comment:
this should be a part of the `tearDown` function so that it automatically gets called after each test and we don't need to duplicate code.

Reply:
We don't need to delete the path after every function call, utils.save_parquet overwrites anyway. I will remove all the calls but thanks for introducing me to the function, will be very useful in time :)

Reply:
I still think making tests have no side effects makes sense, because some other test in some other file expects that directory to be empty.

Reply:
I agree but some tests would not want the file to be empty like in recommendations where the results once saved would be used in between the test functions. I have left this part to be handled later so that we are sure when to use teardown, teardown_class, teardown_method or inheriting from a different parent class altogether.

Reply:
OK, but in general, I would like tests to be independent of each other. If a test needs some file, it should create the file itself (or get it from its setup function). It should never expect another test to create the file because the order that tests run in is not guaranteed to be the same between test runs.

Reply:
gotcha!